src/server/pfs/s3/object.go

```diff
@@ -0,0 +1,172 @@
+package s3
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"strings"
+	"fmt"
+	"crypto/md5"
+	"encoding/base64"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/src/client"
+)
+
+type objectHandler struct {
+	pc *client.APIClient
+}
+
+func newObjectHandler(pc *client.APIClient) *objectHandler {
+	return &objectHandler{pc: pc}
+}
+
+func (h *objectHandler) get(w http.ResponseWriter, r *http.Request) {
+	repo, branch, file := objectArgs(w, r)
+	branchInfo, err := h.pc.InspectBranch(repo, branch)
+	if err != nil {
```

Comment:
This error could be something other than a not found right? That's the most likely error but we might not want to assume that's what it is.

Reply:
`notFoundError` checks if the error message implies a 404, and if not, it falls back to a 500. Should I give it a better name?

Reply:
Yeah, that name is a little confusing to me.

Reply:
Renamed to `maybeNotFoundError`, as it was before. I'm not too fond of the name, but can't think of a better one.